File: samples/CameraAccess/CameraAccess/OpenClaw/JarvisChat.swift
import Foundation

struct JarvisChat: Identifiable, Codable {
    let id: String
    var title: String
    var previewText: String
    var createdAt: Date
    var updatedAt: Date
    var sessionKey: String

    static func makeId() -> String {
        String(UUID().uuidString.replacingOccurrences(of: "-", with: "").prefix(8).lowercased())
    }

    static func autoTitle(from text: String) -> String {
        let words = text.split(separator: " ").prefix(5).joined(separator: " ")
        return words.isEmpty ? "New Chat" : words
    }
}

File: samples/CameraAccessAndroid/app/src/main/java/com/meta/wearable/dat/externalsampleapps/cameraaccess/openclaw/JarvisChat.kt
package com.meta.wearable.dat.externalsampleapps.cameraaccess.openclaw

import java.util.UUID

data class JarvisChat(
    val id: String,
    var title: String,
    var previewText: String,
    val createdAt: Long = System.currentTimeMillis(),
    var updatedAt: Long = System.currentTimeMillis(),
    val sessionKey: String,
) {
    companion object {
        fun makeId(): String =
            UUID.randomUUID().toString().replace("-", "").take(8).lowercase()

        fun autoTitle(text: String): String {
            val words = text.trim().split("\\s+".toRegex()).take(5).joinToString(" ")
            return words.ifEmpty { "New Chat" }
        }
    }
}
